app/notification.go

```diff
@@ -673,34 +725,32 @@ func (n *postNotification) GetSenderName(userNameFormat string, overridesAllowed
 	return n.sender.GetDisplayNameWithPrefix(userNameFormat, "@")
 }
 
-// addMentionedUsers will add the mentioned user id in the struct's list for mentioned users
-func (e *ExplicitMentions) addMentionedUsers(ids []string) {
-	for _, id := range ids {
-		e.MentionedUserIds[id] = true
-	}
-}
-
 // checkForMention checks if there is a mention to a specific user or to the keywords here / channel / all
 func (e *ExplicitMentions) checkForMention(word string, keywords map[string][]string) bool {
 	isMention := false
 
+	mentionType := KeywordMention
+
 	switch strings.ToLower(word) {
 	case "@here":
 		e.HereMentioned = true
+		mentionType = ChannelMention
 	case "@channel":
 		e.ChannelMentioned = true
+		mentionType = ChannelMention
 	case "@all":
 		e.AllMentioned = true
+		mentionType = ChannelMention
 	}
 
 	if ids, match := keywords[strings.ToLower(word)]; match {
-		e.addMentionedUsers(ids)
+		e.addMentions(ids, mentionType)
 		isMention = true
```

Comment:
shouldn't we have a return here? if this matches, do we want to go to the other case being the same mention type?